PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['event.event']
-    _inherit = ['event.event.ticket']
-    _inherit = ['event.type.ticket']
-    _inherit = ['product.product']
-    _inherit = ['product.template']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/event_product/models/event_event.py
+++ b/addons/event_product/models/event_event.py
@@ -2,7 +2,7 @@ from odoo import fields, models
 
 
 class EventEvent(models.Model):
-    _inherit = ['event.event']
+    _inherit = 'event.event'
 
     currency_id = fields.Many2one(
         'res.currency', string='Currency',

--- a/addons/event_product/models/event_event_ticket.py
+++ b/addons/event_product/models/event_event_ticket.py
@@ -3,7 +3,7 @@ from odoo import api, models, fields
 
 
 class EventEventTicket(models.Model):
-    _inherit = ['event.event.ticket']
+    _inherit = 'event.event.ticket'
     _order = "event_id, sequence, price, name, id"
 
     price_reduce_taxinc = fields.Float(

--- a/addons/event_product/models/event_type_ticket.py
+++ b/addons/event_product/models/event_type_ticket.py
@@ -9,7 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class EventTypeTicket(models.Model):
-    _inherit = ['event.type.ticket']
+    _inherit = 'event.type.ticket'
     _order = "sequence, price, name, id"
 
     def _default_product_id(self):

--- a/addons/event_product/models/product_product.py
+++ b/addons/event_product/models/product_product.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class ProductProduct(models.Model):
-    _inherit = ['product.product']
+    _inherit = 'product.product'
 
     event_ticket_ids = fields.One2many('event.event.ticket', 'product_id', string='Event Tickets')

--- a/addons/event_product/models/product_template.py
+++ b/addons/event_product/models/product_template.py
@@ -2,7 +2,7 @@ from odoo import fields, models
 
 
 class ProductTemplate(models.Model):
-    _inherit = ['product.template']
+    _inherit = 'product.template'
 
     service_tracking = fields.Selection(selection_add=[
         ('event', 'Event Registration'),
